autoload/wiki/complete.vim

```diff
@@ -29,7 +29,8 @@ function! wiki#complete#link(lead, line, pos) abort " {{{1
   let l:candidates = s:completer_wikilink.complete(l:input)
   call map(l:candidates, 'v:val.word')
   if !s:completer_wikilink.is_anchor
-    return filter(l:candidates, 'stridx(v:val, l:input) == 0')
+    return filter(l:candidates, {_,x -> match(x,
+          \ (g:wiki_completion_case_sensitive ? '\C' : '\c') . l:input) == 0})
```

Comment:
This is _probably_ OK, but note that regex matching is usually much slower than the `stridx`. One possibility here would be to only do regex matching if we are case insensitive. On the other hand, premature optimization is not really necessary...

Reply:
Just for fun, I threw together a test of `stridx` vs `match` (see [this gist](https://gist.github.com/anjiro/db5b539242d6791c46c59fb6025b9874)). It generates a list with 100,000 100-character strings, then calls `filter` with a random 10-character string, using `stridx` or `match`. `stridx` is indeed faster than `match` - by 0.25s on my iPad, and by .19s on my MacBook Pro. So I think we’re safe!

Reply:
Thanks! for testing - it confirms my understanding, but it also shows that the difference is insignificant.